src/cascadia/TerminalApp/AppActionHandlers.cpp

```diff
@@ -581,4 +583,89 @@ namespace winrt::TerminalApp::implementation
         }
     }
 
+    // Function Description:
+    // - Helper to launch a new WT instance. It can either launch the instance
+    //   elevated or unelevated.
+    // - To launch elevated, it will as the shell to elevate the process for us.
+    //   This might cause a UAC prompt. The elevation is performed on a
+    //   background thread, as to not block the UI thread.
+    // Arguments:
+    // - elevate: If true, launch the new Terminal elevated using `runas`
+    // - newTerminalArgs: A NewTerminalArgs describing the terminal instance
+    //   that should be spawned. The Profile should be filled in with the GUID
+    //   of the profile we want to launch.
+    // Return Value:
+    // - <none>
+    // Important: Don't take the param by reference, since we'll be doing work
+    // on another thread.
+    fire_and_forget _OpenNewWindow(const bool elevate,
+                                   const NewTerminalArgs newTerminalArgs)
+    {
+        // Hop to the BG thread
+        co_await winrt::resume_background();
+
+        // This will get us the correct exe for dev/preview/release. If you
+        // don't stick this in a local, it'll get mangled by ShellExecute. I
+        // have no idea why.
+        const auto exePath{ GetWtExePath() };

```

Comment:
technically only works for CascadiaPackage and not WTU

Reply:
if we were doing this to the gills we would propagate the event out of AppLogic and to the win32 host so the win32 host could spawn another copy of itself

Reply:
this is not a blocking comment!